File: src/main/java/phoenix/integration/minecraft/PhoenixRecipeBookCategories.java
package phoenix.integration.minecraft;

import net.minecraft.client.util.RecipeBookCategories;
import net.minecraft.item.ItemStack;
import net.minecraft.item.Items;
import phoenix.init.PhoenixBlocks;
import phoenix.utils.EnumUtil;

public class PhoenixRecipeBookCategories
{
    public static final RecipeBookCategories MECHANISMS = EnumUtil.addEnum(RecipeBookCategories.class,
            "MECHANISMS", new Class[]{ItemStack[].class}, (Object) new ItemStack[]{new ItemStack(Items.COMPARATOR)});
    public static final RecipeBookCategories OVEN = EnumUtil.addEnum(RecipeBookCategories.class,
            "OVEN", new Class[]{ItemStack[].class}, (Object) new ItemStack[]{new ItemStack(PhoenixBlocks.INSTANCE.getOVEN().get())});

}

File: src/main/kotlin/phoenix/integration/minecraft/PhoenixRecipeBookCategories.kt
package phoenix.integration.minecraft

import net.minecraft.client.util.RecipeBookCategories
import net.minecraft.item.ItemStack
import net.minecraft.item.Items
import phoenix.init.PhxBlocks
import phoenix.utils.EnumUtil

object PhoenixRecipeBookCategories
{
    val MECHANISMS = EnumUtil.addEnum(RecipeBookCategories::class.java, "MECHANISMS", arrayOf<Class<*>>(Array<ItemStack>::class.java), arrayOf(ItemStack(Items.COMPARATOR)) as Any)
    val OVEN       = EnumUtil.addEnum(RecipeBookCategories::class.java, "oven", arrayOf<Class<*>>(Array<ItemStack>::class.java), arrayOf(ItemStack(PhxBlocks.oven)) as Any)
}
